infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/api/server/impls/KubernetesNamespaceMetaImpl.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.api.server.impls;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.eclipse.che.workspace.infrastructure.kubernetes.api.shared.KubernetesNamespaceMeta;
+
+/** @author Sergii Leshchenko */
+public class KubernetesNamespaceMetaImpl implements KubernetesNamespaceMeta {
+
+  private String name;
+  private Map<String, String> attributes;
+
+  public KubernetesNamespaceMetaImpl(String name) {
+    this.name = name;
+  }
+
+  public KubernetesNamespaceMetaImpl(String name, Map<String, String> attributes) {
+    this.name = name;
+    if (attributes != null) {
+      this.attributes = new HashMap<>(attributes);
+    }
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  @Override
+  public Map<String, String> getAttributes() {
+    if (attributes == null) {
+      attributes = new HashMap<>();
+    }
+    return attributes;
+  }
+
+  public void setAttributes(Map<String, String> attributes) {
+    this.attributes = attributes;
```

Comment:
We're creating a copy of attributes in the constructor, yet we directly use the supplied instance here. What is the correct approach?

Reply:
It's not documented. So, nobody knows )
But it's POJO style that we have in other places, like fundamental Workspace entity https://github.com/eclipse/che/blob/caae00e21b2fd316560a406f2e1d8ad98edf1dad/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/model/impl/WorkspaceImpl.java#L269

It makes sense to contribute some contribution guide that would hold some info like that.

Reply:
I guess it doesn't make much difference because we're  not using the map setters too much, but rather use `getX().put*()` and when we do use the map setter, we're disciplined about not leaking the map out of the context where the setter is used. It's just something that we need to keep an eye on.